batch/test/advanced_creation.js

```diff
@@ -0,0 +1,276 @@
+const {customAlphabet} = require('nanoid');
+const allowedChars = 'qwertyuiopasdfghjklzxcvbnm';
+const testRunId = customAlphabet(allowedChars, allowedChars.length)();
+
+const execSync = cmd => cp.execSync(cmd, {encoding: 'utf-8'});
+
+const cwd = path.join(__dirname, '..');
+
+describe('Create jobs with container, template and bucket', () => {
+  let projectId;
+  let bucketName;
+  let templateName;
+
+  before(async () => {
+    projectId = await batchClient.getProjectId();
+    bucketName = `batch-js-test-bucket-${testRunId}`;
+    templateName = `batch-js-test-template-${testRunId}`;
+    await createBucket(bucketName);
+    await createTemplate(projectId, templateName);
+  });
+
+  it('create a job with a container payload', async () => {
+    execSync(
+      `node create/create_with_container_no_mounting.js ${projectId} us-central1 test-job-js-container-${testRunId}`,
+      {cwd}
+    );
+  });
+
+  it('create a job with a GCS bucket', async () => {
+    execSync(
+      `node create/create_with_mounted_bucket.js ${projectId} us-central1 test-job-js-bucket-${testRunId} ${bucketName}`,
+      {cwd}
+    );
+  });
+
+  it('create a job with instance template', async () => {
+    execSync(
+      `node create/create_script_job_with_template.js ${projectId} us-central1 test-job-js-template-${testRunId} ${templateName}`,
+      {cwd}
+    );
+  });
+
+  // waiting for jobs to succed in separate tests lets us create them all on the server and let them run in parallel,
+  // so the tests complete multiple times faster
+
+  it('wait for a job with a GCS bucket to succeed', async () => {
+    await waitForJobToSucceed(
+      projectId,
+      'us-central1',
+      `test-job-js-bucket-${testRunId}`
+    );
+  });
+
+  it('wait for a job with a container payload to succeed', async () => {
+    await waitForJobToSucceed(
+      projectId,
+      'us-central1',
+      `test-job-js-container-${testRunId}`
+    );
+  });
+
+  it('wait for a job with an instance template to succeed', async () => {
+    await waitForJobToSucceed(
+      projectId,
+      'us-central1',
+      `test-job-js-template-${testRunId}`
+    );
+  });
+
+  after(async () => {
+    await cleanupBucket(bucketName);
+    await deleteInstanceTemplate(projectId, templateName);
+  });
+});
+
+async function createBucket(bucketName) {
+  // For default values see: https://cloud.google.com/storage/docs/locations and
+  // https://cloud.google.com/storage/docs/storage-classes
+
+  await storage.createBucket(bucketName, {
+    location: 'US-CENTRAL1',
+    storageClass: 'COLDLINE',
+  });
+}
+
+async function cleanupBucket(bucketName) {
+  // GCS bucket will fail to delete if it is non-empty.
+  // There is no 'force=true' argument in the Node GCS client,
+  // so we cannot use force deletion here and have to first delete all files,
+  // and only then delete the actual bucket.
+  for (let i = 0; i <= 3; i++) {
+    await deleteFileInBucket(bucketName, `output_task_${i}.txt`);
```

Comment:
You might consider taking all of these promises and doing `await Promise.all()` so it doesn't block for each delete. (The deleteBucket of course would still need to be last.)